tiledb/sm/query/query.cc

```diff
@@ -1054,6 +1082,18 @@ Status Query::set_subarray_unsafe(const NDRange& subarray) {
   return Status::Ok();
 }
 
+const Subarray& Query::subarray() const {
+  if (type_ == QueryType::WRITE)
+    return *writer_.subarray_ranges();
+  return *reader_.subarray();
+}
+
+Subarray* Query::subarray() {
+  if (type_ == QueryType::WRITE)
+    return const_cast<Subarray*>(writer_.subarray_ranges());
+  return const_cast<Subarray*>(reader_.subarray());
```

Comment:
Modify the Reader/Writer APIs to return a non-const `Subarray`, avoid `const_cast` at all costs.

Reply:
const_cast removed.